### PR TITLE
Isabella branch for Protobuf 

### PR DESCRIPTION
**Pull request by Isabella373**
### Length Justification and Key Files to Review

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

**Review by nimazareian — COMMENTED** [review]
LGTM! :100: Just pending the merge conflict and the failing CI (should get fixed once you pull from master branch)

File: src/software/sensor_fusion/sensor_fusion.h [hunk 1]
```diff
@@ -82,7 +82,8 @@ class SensorFusion
      *
      * @return Ball if filtered from ball detections
      */
-    std::optional<Ball> createBall(const std::vector<BallDetection> &ball_detections);
+    std::optional<Ball> createBallProto(
+        const std::vector<BallDetection> &ball_detections);
```
Comment (itsarune) [inline src/software/sensor_fusion/sensor_fusion.h]: `Ball` [isn't a proto type,](https://github.com/UBC-Thunderbots/Software/blob/master/src/software/world/ball.h) this should remain as `createBall`

File: src/software/test_util/test_util.cpp [hunk 2]
```diff
@@ -131,8 +132,8 @@ namespace TestUtil
         return states;
     }
 
-    GameState createGameState(const RefereeCommand &current_referee_command,
-                              const RefereeCommand &previous_referee_command)
+    GameState createGameStateProto(const RefereeCommand &current_referee_command,
```
Comment (itsarune) [inline src/software/test_util/test_util.cpp]: GameState is a c++ type, not a Proto type (a proto is any file found in software/proto), this should remain as the same

[GameState](https://github.com/UBC-Thunderbots/Software/blob/master/src/software/world/game_state.h)

File: src/software/test_util/test_util.h [hunk 3]
```diff
@@ -173,8 +173,8 @@ namespace TestUtil
      * @param previous_referee_command The name of the previous referee command to set
      * @return the new GameState
      */
-    GameState createGameState(const RefereeCommand &current_referee_command,
-                              const RefereeCommand &previous_referee_command);
+    GameState createGameStateProto(const RefereeCommand &current_referee_command,
```
Comment (itsarune) [inline src/software/test_util/test_util.h]: GameState is a c++ type, not a Proto type (a proto is any file found in software/proto), this should remain as the same

[GameState](https://github.com/UBC-Thunderbots/Software/blob/master/src/software/world/game_state.h)

File: src/software/world/field.cpp [hunk 4]
```diff
@@ -20,7 +20,7 @@ Field Field::createSSLDivisionAField()
     return field;
 }
 
-Field Field::createField(TbotsProto::FieldType field_type)
+Field Field::createFieldProto(TbotsProto::FieldType field_type)
```
Comment (itsarune) [inline src/software/world/field.cpp]: this is returning a C++ Field object (not a proto that is generated by the protobuf compiler).

File: src/software/world/field.h [hunk 5]
```diff
@@ -55,7 +55,7 @@ class Field
      * @param field_type The field type
      * @return a field with the standard SSL Division A or B dimensions
      */
-    static Field createField(TbotsProto::FieldType field_type);
+    static Field createFieldProto(TbotsProto::FieldType field_type);
```
Comment (itsarune) [inline src/software/world/field.h]: this is returning a C++ type, not a Proto type

**Review by itsarune — CHANGES_REQUESTED** [review]
Good start!

I think there is a bit of a confusion here. We want to rename functions that create a Protobuf type. A Protobuf type is automatically generated by the C++ protobuf compiler and in our codebase, we store them here: https://github.com/UBC-Thunderbots/Software/tree/master/src/proto

In our code, these Proto types are seen as "TbotsProto::PassVisualization", "TbotsProto::BallState", etc. We want to rename functions that return these proto-based types.

It is confusing that sometimes we mimic these proto names to create equivalent c++ classes (BallState [ball_state.h], Ball [ball.h], GameState [game_state.h] etc.). These are c++ objects and these shouldn't be renamed.

So if a function returns: TbotsProto::GameState -> rename the function to `createGameStateProto`.
Likewise, if a function returns GameState -> this function should be called `createGameState`

**Review by jameschuang2002 — COMMENTED** [review]
I have the same comment as Arun. I think we should follow the old convention with protobuf naming

**Pull request closed without merging**